### PR TITLE
Added echo_server code

### PR DESCRIPTION
**Pull request by pinglinh**
Please add comments on how to improve this code.

### CODE REVIEW

File: echo_server.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+class UserEcho
+
+	def initialize(welcome, instructions)
+		@welcome = "Welcome!"
+		@instructions = "Please enter a word to be reprinted, to exit this program please enter exit:"
+	end
+
+	def welcome_message
+		@welcome
+	end
```
Comment (pinglinh) [inline echo_server.rb]: testing

File: echo_server.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+class UserEcho
+
+	def initialize(welcome, instructions)
```
Comment (cnak) [inline echo_server.rb]: Unused input parameters. - "welcome" and "instructions"

File: echo_server.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	def instruction_message
+		@instructions
+	end
+
+	def start
+		@welcome
+		@instructions
+		input
+	end
+
```
Comment (cnak) [inline echo_server.rb]: Think about your public facing interface.

Perhaps the below methods could be private?

File: echo_server.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	def input
+		@@user_input = STDIN.readline.chomp
+		exit_entered
+	end
+
```
Comment (cnak) [inline echo_server.rb]: Not sure about this "more_input" method.  :/

Doesn't the "input" method already does this?

File: echo_server.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	def more_input
+		input
+	end
+
```
Comment (cnak) [inline echo_server.rb]: Not sure why @@user_input is a class variable

File: echo_server.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
+class UserEcho
+
+	def initialize(welcome, instructions)
+		@welcome = "Welcome!"
+		@instructions = "Please enter a word to be reprinted, to exit this program please enter exit:"
+	end
+
```
Comment (cnak) [inline echo_server.rb]: Perhaps it'll be easier to read if you group related methods together.

welcome_message
instruction_message
goodbye_message

File: echo_server.rb [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+	def more_input
+		input
+	end
+
+	def exit
+		@@user_input == "exit"
+	end
+
```
Comment (cnak) [inline echo_server.rb]: exit entered reads a bit awkwardly.

"while exit is not false do ....."  I personally prefer "while exit is true do this"

File: echo_server.rb (hunk 3 above)
Comment (cnak) [inline echo_server.rb]: Good small start method

**Review by cnak — CHANGES_REQUESTED** [review]
Please look at the comments above.

File: echo_server.rb (hunk 2 above)
Comment (pinglinh) [reply echo_server.rb]: Hmm not sure how I would have it printed first when the server starts?

**Pull request merged**